packages/salesforcedx-utils/src/cli/checkCliVersion.ts

```diff
@@ -0,0 +1,125 @@
+      return sfVersion;
+    } catch {
+      return 'No SF CLI';
+    }
+  }
+
+  public async parseSfdxCliVersion(sfdxCliVersion: string): Promise<number[]> {
+    // Both SFDX and SF v2 can match a valid SFDX version, so there are 2 patterns to search for
+    const sfdxPattern = /sfdx-cli\/(\d+\.\d+\.\d+)/;
+    const sfdxMatch = sfdxPattern.exec(sfdxCliVersion);
+    const sfPattern = /@salesforce\/cli\/(\d+\.\d+\.\d+)/;
+    const sfMatch = sfPattern.exec(sfdxCliVersion);
+    if (sfdxMatch) {
+      const sfdxVersion = sfdxMatch[1];
+      const sfdxVersionNumber = sfdxVersion.split('.').map(Number);
+      return sfdxVersionNumber;
+    }
+    else if (sfMatch) {
+      const sfdxVersion = sfMatch[1];
+      const sfdxVersionNumber = sfdxVersion.split('.').map(Number);
+      return sfdxVersionNumber;
+    }
+    return [-1];
+  }
+
+  public async parseSfCliVersion(sfCliVersion: string): Promise<number[]> {
+    const sfPattern = /@salesforce\/cli\/(\d+\.\d+\.\d+)/;
+    const sfMatch = sfPattern.exec(sfCliVersion);
+    if (sfMatch) {
+      const sfVersion = sfMatch[1];
+      const sfVersionNumber = sfVersion.split('.').map(Number);
+      return sfVersionNumber;
+    }
+    return [-1];
+  }
+
+  public async validateCliInstallationAndVersion(sfdxCliVersionArray: number[], sfCliVersionArray: number[]): Promise<Number> {
+    // The last working version of SFDX is v7.193.2
+    const minSFDXVersion = '7.193.2';
+    const minSFDXVersionArray = minSFDXVersion.split('.').map(Number);
+
+    // SF v1 does not map sf commands to sfdx commands and is not supported
+    const minSFVersion = '2.0.0';
+    const minSFVersionArray = minSFVersion.split('.').map(Number);
+
+    // Case 1: Neither SFDX CLI nor SF CLI is installed
+    if (sfdxCliVersionArray[0] === -1 && sfCliVersionArray[0] === -1) {
+      return CheckCliEnum.cliNotInstalled;
+    }
+
+    // Case 2: Only SF CLI (v1) is installed (SF v1 cannot be used because it does not map sf to sfdx)
+    if (sfdxCliVersionArray[0] === -1 && sfCliVersionArray[0] === 1) {
+      return CheckCliEnum.onlySFv1;
+    }
+
+    // Case 3: Both SFDX CLI (v7) and SF CLI (v2) are installed at the same time
+    if (sfCliVersionArray[0] >= minSFVersionArray[0] && !sfdxCliVersionArray.every((val, index) => val === sfCliVersionArray[index])) {
+      return CheckCliEnum.bothSFDXAndSFInstalled;
+    }
+
+    // Case 4: SFDX CLI is installed - need to validate if the version is above the minimum supported version
+    if (sfCliVersionArray[0] < minSFVersionArray[0] && sfdxCliVersionArray[0] !== -1) {
+      if (
+        sfdxCliVersionArray[0] >= minSFDXVersionArray[0] &&
+        sfdxCliVersionArray[1] >= minSFDXVersionArray[1] &&
+        sfdxCliVersionArray[2] >= minSFDXVersionArray[2]
+      ) {
+        return CheckCliEnum.validCli;
+      } else {
+        return CheckCliEnum.outdatedSFDXVersion;
+      }
+    }
+
+    // Case 5: SF v2 is installed
+    return CheckCliEnum.validCli;
```

Comment:
I didn't do another check for SF versions because for SF v2 anything above 2.0.0 is supported.  The case of only SF v1 has already been caught in Case 2.